sdk/src/program-manager.ts

```diff
@@ -357,7 +355,7 @@ class ProgramManager {
         console.log("Running program offline")
         console.log("Proving key: ", provingKey);
         console.log("Verifying key: ", verifyingKey);
-        return this.executionEngine.executeFunctionOffline(executionPrivateKey, program, function_name, inputs, proveExecution,false, imports, provingKey, verifyingKey);
+        return WasmProgramManager.executeFunctionOffline(executionPrivateKey, program, function_name, inputs, proveExecution, false, imports, provingKey, verifyingKey);
```

Comment:
Shouldn't this remove the `false` cache argument as well?

Reply:
The old `synthesize_keypair` argument calling the `synthesize_kepair` pathway in SnarkVM created an a complicated monormrophization of the `execution_function` path which added ~1MB to the wasm binary size. The new pathway uses the execute_function_offline function which returns an optional object with the synthesized key in it within the `ExecutionResponse`

Reply:
How does that affect monomorphization? The cache doesn't change the static type, and it's always passing `false` anyways.

Reply:
The cache within the `ProgramManager` had no effect on the `wasm` binary size - it was removed because it was ultimately a poor caching strategy. 

What had an effect on it was calling invoking the `synthesize_keypair` method from `SnarkVM` which imported a lot of generics we didn't use directly except for that function. In total removing it took 1MB off the size of the `wasm` binary.

However, SDK users would optimally still like to pre-synthesize keys for storage, so the work is delegated now to the offline execution function, and if `cache` is set to true, then they can get the keys from the function.

Reply:
I see, so you're saying that we *will* use the cache eventually, even though it's currently not being used?

Reply:
Basically it puts the caching onto the implementor. It is being used now in the `synthesize_key` function.

Reply:
Ahhh, I see, I had missed that.